src/panels/lovelace/editor/hui-dialog-edit-card.ts

```diff
@@ -59,42 +67,104 @@ export class HuiDialogEditCard extends LitElement {
         paper-dialog {
           width: 650px;
         }
+        .element-editor {
+          margin-bottom: 16px;
+        }
       </style>
       <paper-dialog with-backdrop>
         <h2>Card Configuration</h2>
         <paper-dialog-scrollable>
-          <hui-yaml-editor
-            .yaml="${this._cardConfig}"
-            @yaml-changed="${this._handleYamlChanged}"
-          ></hui-yaml-editor>
+          ${
+            // Look to see if this has already be done if yes dont use until just render this._elementConfig
+            this._showUIEditor && !this._elementConfig
+              ? html`<div class="element-editor">${until(
+                  this._loadConfig().then(() => this._loadElementConfig()),
```

Comment:
Our render method never should have side effects like calling websocket commands etc. Those things should happen in the showDialog function.

Reply:
I couldnt get the the elementConfig to load before rendering so that the first thing you see is the UI editor not the YAML editor. But I maybe able to use the same method in show dialog using the `then` function. Will work on that